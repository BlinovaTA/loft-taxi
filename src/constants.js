export const PAGES = {
  map: { key: 'map', name: 'Карта', link: '/map' },
  profile: { key: 'profile', name: 'Профиль', link: '/profile' },
  login: { key: 'login', name: 'Выйти', link: '/' }
};

export const PAGES_KEYS = Object.keys(PAGES);