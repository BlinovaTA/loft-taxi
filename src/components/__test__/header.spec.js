import React from "react";
import Header from "../header";
import { PAGES } from '../../constants';
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";

describe("Header", () => {
  describe("when clicked on navigation buttons", () => {
    it("map active", () => {
      const mockStore = {
        getState: () => ({ authorization: { token: 'TOKEN' } }),
        subscribe: () => { },
        dispatch: () => { },
      };
      const history = createMemoryHistory();

      const map = renderer.create(
        <Router history={history}>
          <Provider store={mockStore}>
            <Header page={PAGES.map.key} />
          </Provider>
        </Router>
      ).toJSON();

      expect(map).toMatchSnapshot();
    });

    it("profile active", () => {
      const mockStore = {
        getState: () => ({ authorization: { token: 'TOKEN' } }),
        subscribe: () => { },
        dispatch: () => { },
      };
      const history = createMemoryHistory();

      const profile = renderer.create(
        <Router history={history}>
          <Provider store={mockStore}>
            <Header page={PAGES.profile.key} />
          </Provider>
        </Router>
      ).toJSON();

      expect(profile).toMatchSnapshot();
    });

    it("login active", () => {
      const mockStore = {
        getState: () => ({ authorization: { token: 'TOKEN' } }),
        subscribe: () => { },
        dispatch: () => { },
      };
      const history = createMemoryHistory();

      const login = renderer.create(
        <Router history={history}>
          <Provider store={mockStore}>
            <Header page={PAGES.login.key} />
          </Provider>
        </Router>
      ).toJSON();

      expect(login).toMatchSnapshot();
    });
  });
});
