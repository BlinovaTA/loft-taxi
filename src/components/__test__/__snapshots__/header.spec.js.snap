// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header when clicked on navigation buttons login active 1`] = `
<header
  className="header"
>
  <div
    className="header__logo"
  >
    <img
      alt="logo"
      className="logo__img"
      src="logo.svg"
    />
  </div>
  <ul
    className="header__menu menu"
  >
    <li>
      <a
        className="menu__item "
        href="/map"
        id="map"
        onClick={[Function]}
      >
        Карта
      </a>
    </li>
    <li>
      <a
        className="menu__item "
        href="/profile"
        id="profile"
        onClick={[Function]}
      >
        Профиль
      </a>
    </li>
    <li>
      <a
        className="menu__item  menu__item--active"
        href="/"
        id="login"
        onClick={[Function]}
      >
        Выйти
      </a>
    </li>
  </ul>
</header>
`;

exports[`Header when clicked on navigation buttons map active 1`] = `
<header
  className="header"
>
  <div
    className="header__logo"
  >
    <img
      alt="logo"
      className="logo__img"
      src="logo.svg"
    />
  </div>
  <ul
    className="header__menu menu"
  >
    <li>
      <a
        className="menu__item  menu__item--active"
        href="/map"
        id="map"
        onClick={[Function]}
      >
        Карта
      </a>
    </li>
    <li>
      <a
        className="menu__item "
        href="/profile"
        id="profile"
        onClick={[Function]}
      >
        Профиль
      </a>
    </li>
    <li>
      <a
        className="menu__item "
        href="/"
        id="login"
        onClick={[Function]}
      >
        Выйти
      </a>
    </li>
  </ul>
</header>
`;

exports[`Header when clicked on navigation buttons profile active 1`] = `
<header
  className="header"
>
  <div
    className="header__logo"
  >
    <img
      alt="logo"
      className="logo__img"
      src="logo.svg"
    />
  </div>
  <ul
    className="header__menu menu"
  >
    <li>
      <a
        className="menu__item "
        href="/map"
        id="map"
        onClick={[Function]}
      >
        Карта
      </a>
    </li>
    <li>
      <a
        className="menu__item  menu__item--active"
        href="/profile"
        id="profile"
        onClick={[Function]}
      >
        Профиль
      </a>
    </li>
    <li>
      <a
        className="menu__item "
        href="/"
        id="login"
        onClick={[Function]}
      >
        Выйти
      </a>
    </li>
  </ul>
</header>
`;
